cou/steps/plan.py

```diff
@@ -107,7 +107,8 @@ async def create_upgrade_group(
     for app in filter(filter_function, apps):
         try:
             app_upgrade_plan = app.generate_upgrade_plan(target)
-            group_upgrade_plan.add_step(app_upgrade_plan)
+            if app_upgrade_plan:
```

Comment:
Is this if even needed? Empty steps will not be show to user and their `run` will do nothing.